be/src/io/fs/broker_file_system.cpp

```diff
@@ -415,6 +415,7 @@ Status BrokerFileSystem::download_impl(const Path& remote_file, const Path& loca
 
         RETURN_IF_ERROR(local_writer->append({read_buf.get(), read_len}));
     } // file_handler should be closed before calculating checksum
+    RETURN_IF_ERROR(local_writer->close());
```

Comment:
we may just `return local_writer->close()`
line 420 can be ommited

Reply:
fixed